File: benchmarks/y_startup/probe.go
// Deliberately scoped dispatch probe, not a replacement for Y.
package main

import (
	"os"
	"os/exec"
)

func main() {
	args := os.Args[1:]
	if len(args) == 1 && args[0] == "noop" {
		return
	}
	if len(args) != 2 || args[0] != "focus" {
		os.Exit(2)
	}
	directions := map[string]string{"left": "west", "right": "east", "up": "north", "down": "south"}
	direction, ok := directions[args[1]]
	if !ok {
		os.Exit(2)
	}
	_, err := exec.Command(os.Getenv("Y_BENCH_YABAI"), "-m", "window", "--focus", direction).Output()
	if err != nil {
		if exit, ok := err.(*exec.ExitError); ok {
			os.Exit(exit.ExitCode())
		}
		os.Exit(1)
	}
}

File: benchmarks/y_startup/probe.rs
//! Deliberately scoped dispatch probe, not a replacement for Y.
use std::{
    env,
    process::{self, Command},
};

fn main() {
    let args: Vec<String> = env::args().skip(1).collect();
    if args == ["noop"] {
        return;
    }
    if args.len() != 2 || args[0] != "focus" {
        process::exit(2);
    }
    let direction = match args[1].as_str() {
        "left" => "west",
        "right" => "east",
        "up" => "north",
        "down" => "south",
        _ => process::exit(2),
    };
    let output = Command::new(env::var("Y_BENCH_YABAI").expect("fixture required"))
        .args(["-m", "window", "--focus", direction])
        .output()
        .expect("fixture execution failed");
    process::exit(output.status.code().unwrap_or(1));
}
